### DISCUSSION TITLE
Extracting transition probabilities and confidence intervals

### DISCUSSION THREAD

**Discussion by garneill** [bmcclintock/momentuHMM#78] [Q&A] [answered]
Hi Brett, 

I am using your adaption of Enrico Pirotta's 6-state HMM (Northern Fulmar example) on seabird data and have since adapted the code to add an additional state. We are now looking at the effects of sex, tagging year etc on the transition probabilities. Is it possible to extract the transition probabilities and their confidence intervals from the models to look at the effect of covariates, rather than just the means that are shown when you print the model? I see that these are calculated and plotted within 'plotTPM', but with so many states and the covariates it would be great to extract the values. Is there a function that I have overlooked for extracting these values?

All the best, 
Gavin

**Comment by bmcclintock** [comment, marked as answer]
Hi Gavin,

As of momentuHMM v1.5.0, there are two ways to calculate state transition probabilities as a function of covariates. The function `CIreal` can be used for a specific combination of covariate values from a fitted model. For example, for a fitted 2-state model (`example$m`, which is loaded with the package) with two covariates (`cov1` and `cov2`):

`trProb <- CIreal(example$m, alpha = 0.95, covs = data.frame(cov1=0.5, cov2=0.02), parms = "gamma")`

returns the state transition probability estimates, standard errors (SEs), and 95% confidence intervals (CIs) for these specific covariate values. Alternatively, the function `getTrProbs` can be used to calculate the state transition probability estimates, SEs, and CIs for all (or a subset using the `covIndex` argument) of the observations in the data:

`trProbs <- getTrProbs(example$m, getCI=TRUE)`

See `?CIreal` and `?getTrProbs` for more details and examples.

Cheers,
Brett
